server/py/services/alerts/crud/alerts.py

```diff
@@ -101,16 +101,13 @@ def store_alert(
             )
 
         # if the alert already exists we should check if it should be reset or not
-        if existing_alert is not None and self._should_reset_alert(
-            existing_alert, alert_data, force_reset
-        ):
-            logger.debug(
-                "Resetting alert due to %s",
-                "force_reset being True"
-                if force_reset
-                else "changes in entities, criteria, or trigger of the alert",
+        if existing_alert is not None:
+            should_reset, reset_reason = self._should_reset_alert(
+                existing_alert, alert_data, force_reset
             )
-            self.reset_alert(session, project, new_alert.name)
+            if should_reset:
+                logger.debug("Resetting alert due to %s", reset_reason)
```

Comment:
What alert are we resetting?
Also, it's better to use arguments over formatted strings, as the logs stay consistent.
```suggestion
                logger.debug("Resetting alert before storing", project=project, alert_name=name, reason=reset_reason)
```